lib/graphql/analysis/analyze_query.rb

```diff
@@ -21,7 +26,11 @@ def analyze_query(query, analyzers)
         reduce_node(op_node, analyzers_and_values)
       end
 
-      analyzers_and_values.map { |(r, value)| finalize_reducer(r, value) }
+      if !@errors.blank?
+        @errors.flatten
+      else
+        analyzers_and_values.map { |(r, value)| finalize_reducer(r, value) }
```

Comment:
In this flow, if _any_ reducer causes an error, then _none_ of the reducers are finalized. Is that the right behavior? 

For example, let's say you have two reducers: 

- one of them measures incoming query complexity and logs it;
- the other checks for valid permissions and turns away queries when people don't have permission

If someone makes a request with invalid permissions, should we still log the query complexity? 

(I started this comment by assuming the answer was yes, but now I'm not sure: since there was an error, we didn't _actually_ run the query, so maybe we shouldn't log it.)